### PR TITLE
feat/coded errors

### PR DESCRIPTION
**Pull request by khcoaching**
- Adds shared `CodedErrors` to be used in all somm apps
- Adds `CodedHTTPErrors` to be thrown in HTTP routes
- Polling functions execute immediately on the first try
- Add a benchmark that belonged in the pool sorting PR

### CODE REVIEW

File: packages/benchmarks/scripts/big-number-compare.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+const Benchmark = require('benchmark');
```
Comment (khcoaching) [inline packages/benchmarks/scripts/big-number-compare.js:1]: belonged in other pr, whoops

File: packages/server/src/services/bitquery/fetcher.ts [hunk 2]
```diff
@@ -8,7 +8,7 @@ import {
 import fetch from 'cross-fetch';
```
Comment (khcoaching) [inline packages/server/src/services/bitquery/fetcher.ts:8]: @kkennis requires your revew

File: packages/server/src/services/bitquery/fetcher.ts [hunk 3]
```diff
@@ -97,14 +87,31 @@ export default class BitqueryFetcher {
...
+}
 
-        return dexTrades;
+type OHLCParams = {
+    baseTokenId: string,
+    quoteTokenId: string,
+    startDate: string,
+    endDate: string,
+}
+async function getPoolDailyOHLC({ baseTokenId, quoteTokenId, startDate, endDate }: OHLCParams) {
```
Comment (khcoaching) [inline packages/server/src/services/bitquery/fetcher.ts:101]: @kkennis refactored this duplicated logic out

File: packages/server/src/services/bitquery/fetcher.ts [hunk 4]
```diff
@@ -44,15 +44,10 @@ export default class BitqueryFetcher {
 
     static async getLastDayOHLC(baseTokenId: string, quoteTokenId: string): Promise<DexTrade> {
         // Calculate start Date and endDate
-        const endDate = BitqueryFetcher.formatDate(endOfDay(new Date()));
```
Comment (khcoaching) [inline packages/server/src/services/bitquery/fetcher.ts:47]: this was essentially the same as getPeriodOHLC

File: packages/server/src/services/bitquery/fetcher.ts [hunk 5]
```diff
@@ -44,15 +44,10 @@ export default class BitqueryFetcher {
 
     static async getLastDayOHLC(baseTokenId: string, quoteTokenId: string): Promise<DexTrade> {
```
Comment (khcoaching) [inline packages/server/src/services/bitquery/fetcher.ts:45]: this was essentially the same as getPeriodOHLC where start and end were 1 day apart

File: packages/server/src/services/bitquery/fetcher.ts (hunk 2 above)
Comment (khcoaching) [inline packages/server/src/services/bitquery/fetcher.ts:8]: @kkennis this requires your review. I refactored duplicate logic and encapsulated the coded errors

**Comment by khcoaching** [comment]
@kkennis just hit the merge button if it looks good

**Review by kkennis — APPROVED** [review]

**Pull request merged**